fprime-sensors/PiCamera/Components/PiCameraManager/PiCameraManager.cpp

```diff
@@ -0,0 +1,238 @@
+// resolution settings: we're using 1920 x 1080 (HD)
+const U32 PiCameraManager::IMAGE_WIDTH = 1920;
+const U32 PiCameraManager::IMAGE_HEIGHT = 1080;
+
+// Component construction and destruction
+
+// Constructor: called when a component is created (default method)
+PiCameraManager::PiCameraManager(const char* const compName) 
+    : PiCameraManagerComponentBase(compName),
+      m_cameraEnabled(false), // start with the camera disabled
+      m_imageCounter(0), // the # of images starts at 0
+      m_lastImageSize(0) { // last image size starts at 0
+    
+    // Create image directory (if it doesn't exist)
+    // 0755 = permissions: can read, write, execute
+    mkdir(IMAGE_DIR, 0755);
+}
+
+// Destructor: when a component is destroyed
+PiCameraManager::~PiCameraManager() {
+    if (m_cameraEnabled) {
+        shutdownCamera();
+    }
+}
+
+// Command Handlers
+
+// turns the camera on
+void PiCameraManager::ENABLE_CAMERA_cmdHandler(FwOpcodeType opCode, U32 cmdSeq) {
+    // confirm the camera is on
+    if (m_cameraEnabled) {
+        // Emit the camera enabled event
+        this->log_ACTIVITY_LO_CAMERA_ENABLED_EVENT();
+        
+        // Update telemetry channel (reports specific parameters)
+        this->tlmWrite_CAMERA_ENABLED(true);
+        
+        // Send command response
+        this->cmdResponse_out(opCode, cmdSeq, Fw::CmdResponse::OK);
+        return;
+    }
+
+    // Initialises the camera (connects to the initializeCamera method below)
+    if (initializeCamera()) {
+        // it's enabled until proven otherwise
+        m_cameraEnabled = true;
+        
+        // Emit the camera enabled event
+        this->log_ACTIVITY_LO_CAMERA_ENABLED_EVENT();
+        
+        // Update telemetry channel
+        this->tlmWrite_CAMERA_ENABLED(true);
+        
+        // Send command response
+        this->cmdResponse_out(opCode, cmdSeq, Fw::CmdResponse::OK);
+    } else { // if it fails, print the error message
+        Fw::LogStringArg errorStr("Failed to initialize camera");
+        // send the event to GDS with high (HI) priority warning
+        this->log_WARNING_HI_PHOTO_CAPTURE_FAILED(errorStr);
+        // tell F Prime it has failed too
+        this->cmdResponse_out(opCode, cmdSeq, Fw::CmdResponse::EXECUTION_ERROR);
+    }
+}
+
+// Turns the camera off
+void PiCameraManager::DISABLE_CAMERA_cmdHandler(FwOpcodeType opCode, U32 cmdSeq) {
+    // if camera is currently enabled, shut it down
+    if (m_cameraEnabled) {
+        shutdownCamera();
+        m_cameraEnabled = false;
+    }
+    
+    // Emit the camera disabled event
+    this->log_ACTIVITY_LO_CAMERA_DISABLED_EVENT();
+    
+    // Update telemetry channel
+    this->tlmWrite_CAMERA_ENABLED(false);
+    
+    // Send command response
+    this->cmdResponse_out(opCode, cmdSeq, Fw::CmdResponse::OK);
+}
+
+// Takes a photo
+void PiCameraManager::CAPTURE_PHOTO_cmdHandler(FwOpcodeType opCode, U32 cmdSeq) {
```

Comment:
I would recommend adding an argument for the filename. This filename would be used to store the captured photo instead of generating one in the moment.